src/refreshtoken.ts

```diff
@@ -0,0 +1,86 @@
+interface RefreshTokenRenewResponse {
+  status: string,
+  userId: string | null,
+  refreshAfter: string | null,
+}
+
+const REFRESH_TOKEN_API = '/api/vtexid/refreshtoken/webstore'
+const LOCAL_STORAGE_KEY = 'vid_rt_webstore'
+const STATUS = {
+  SUCCESS: 'success',
+  INVALID_SESSION: 'invalidsession',
+  ERROR: 'error'
+}
+const TTL_30_MIN = 30 * 60 * 1000
+const TTL_12_HOURS = 12 * 60 * 60 * 1000
+const APPNAME = 'render-session'
+const TIMEOUT = 5000
+
+const setItem = (status: string, refreshAfter: Date) =>
+  localStorage.setItem(
+    LOCAL_STORAGE_KEY,
+    JSON.stringify({ status, refreshAfter })
+  )
+
+const setError = () =>
+  setItem(STATUS.ERROR, new Date(new Date().getTime() + TTL_30_MIN))
+
+const setInvalidSession = () =>
+  setItem(STATUS.INVALID_SESSION, new Date(new Date().getTime() + TTL_12_HOURS))
+
+const fetchWithTimeout = (url: string, options: RequestInit | undefined, timeout: number): Promise<Response> => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => reject(new Error("timeout")), timeout)
+    fetch(url, options).then(resolve, reject)
+  })
+}
+
+export const renew = async () => {
+  const localStorage = window.localStorage
+  if (!window || !localStorage) return
```

Comment:
`!window` doesn't make much sense. If it's null, you'll get a Null Reference in the first line.